services/discourse_translator/microsoft.rb

```diff
@@ -56,21 +56,29 @@ def self.access_token
 
       if existing_token
         existing_token
-      else
-        if !SiteSetting.translator_azure_subscription_key.blank?
-          response = Excon.post("#{DiscourseTranslator::Microsoft::ISSUE_TOKEN_URI}?Subscription-Key=#{SiteSetting.translator_azure_subscription_key}")
-
-          if response.status == 200 && (response_body = response.body).present?
-            Discourse.redis.setex(cache_key, 8.minutes.to_i, response_body)
-            response_body
-          elsif response.body.blank?
-            raise TranslatorError.new(I18n.t("translator.microsoft.missing_token"))
-          else
-            # The possible response isn't well documented in Microsoft's API so
-            # it might break from time to time.
-            error = JSON.parse(response.body)["error"]
-            raise TranslatorError.new("#{error['code']}: #{error['message']}")
-          end
+      elsif SiteSetting.translator_azure_subscription_key.present?
+        url = "#{DiscourseTranslator::Microsoft::ISSUE_TOKEN_URI}?Subscription-Key=#{SiteSetting.translator_azure_subscription_key}"
+
+        # Congitive Service's multi-service resource requires a region to be specified
+        # https://docs.microsoft.com/en-us/azure/cognitive-services/translator/reference/v3-0-reference#authenticating-with-an-access-token
+        if SiteSetting.translator_azure_region != 'global'
```

Comment:
Maybe add this comment here:

> Aside from global, the valid choices are retrieved from https://docs.microsoft.com/en-us/azure/cognitive-services/translator/reference/v3-0-reference#authenticating-with-a-multi-service-resource

Reply:
Done :) I've added the comment to the site setting.